test/ng/directive/ngRepeatSpec.js

```diff
@@ -1488,7 +1488,13 @@ describe('ngRepeat animations', function() {
     $rootScope.items = ['1','3'];
     $rootScope.$digest();
 
-    item = $animate.queue.shift();
+    while ($animate.queue.length) {
+      item = $animate.queue.shift();
+      if (item.event === 'leave') {
+        break;
+      }
+    }
```

Comment:
Should tests assume that animation events are queued in a certain order? If not, then perhaps the queue should be tested like this:
```javascript
// Test each queue item's event type
while ($animate.queue.length) {
  item = $animate.queue.shift();
  switch (item.element.text()) {
    case '2':
      expect(item.event).toBe('leave');
      break;
    case '3':
      expect(item.event).toBe('move');
      break;
    default:
      expect(item).toBeUndefined();
  }
}
```

Maybe we should add  a Jasmine matcher in the "ngRepeat Animations" block that compares item.event and item.text() values between two arrays? Should this be new GitHub issue?